middleware/event_stream.go

```diff
@@ -51,70 +31,8 @@ func RaiseEvent(next echo.HandlerFunc) echo.HandlerFunc {
 			return nil
 		}
 
-		if c.Get("event_override") != nil {
-			event, ok := c.Get("event_override").(string)
-			if !ok {
-				l.Log.Warnf("Failed to cast event_override from request - ditching post to kafka")
-				return nil
-			}
-
-			l.Log.Infof("Using overridden event_type %v instead of %v", c.Get("event_override"), eventType)
-			eventType = event
-		}
-
 		l.Log.Infof("Raising Event %v", eventType)
 
-		msg, err := json.Marshal(resource)
-		if err != nil {
-			return err
-		}
-
-		// TODO: make this async? Run this in a goroutine that way the
-		// request isn't effectively blocked by kafka .
-		headers := append(getRequestHeaders(c), kafka.Header{Key: "event_type", Value: []byte(eventType)})
-		err = producer.RaiseEvent(eventType, msg, headers)
-		if err != nil {
-			l.Log.Warnf("failed to raise event to kafka: %v", err)
-			return nil
-		}
-
-		return nil
+		return service.RaiseEvent(c, eventType, resource)
```

Comment:
Maybe I should have been more clear. I know we're going to need to pass the context for the headers or just have another helper method that takes the context as an argument.

I was more implying we should leave the `event_override` and other stuff based on the context here in the middleware. Basically all that `service.RaiseEvent()` would do is marshal the resource + post the message. 

---

Maybe we can change the `getRequestHeaders(c)` function to something like `ForwardableHeaders(c)` so it takes in the echo context and returns the slice of headres - that way the new RaiseEvent function would have a signature like:

```go
func RaiseEvent(eventType string, resource m.Resource, headers []kafka.Header) error

// called via
service.RaiseEvent("Source.pause", source, service.ForwardableHeaders(c))
```

The RaiseEvent function would just add the `event_type` header and produce the message.

---

Does that make sense?

Reply:
Absolutely. I implemented those changes. Thanks for the feedback!